selection.cpp

```diff
@@ -0,0 +1,250 @@
+    }
+    ui->comboBox_answers->setCurrentIndex(m_answer[indexOfCurrentQuestion] - 1);
+}
+
+void Selection::print()
+{
+    for(int k = 0; k < m_size; k++)
+    {
+        if(m_answer[k] == quiz.getCollectionOfQuestions()[k].getCorrectAnswer())
+        {
+            m_correctAnswers++;
+        }
+    }
+
+    ui->stackedWidget->setCurrentIndex(1);
+    ui->label_correctQuestions->setText("Takmicar " + m_playerName + " je pogodio " + QString::number(m_correctAnswers) + "/"
+                                + QString::number(m_size) + " pitanja.");
+    QPixmap picture3("://database/purple-confetti");
+    w = ui->label_confetti->width();
+    h = ui->label_confetti->height();
+    ui->label_confetti->setPixmap(picture3.scaled(w, h, Qt::IgnoreAspectRatio));
+
+    ui->tableWidget->setRowCount(m_size);
+    ui->tableWidget->setColumnCount(2);
+    QStringList hLabels;
+    hLabels << "Pitanje" << "Odgovor";
+    ui->tableWidget->setHorizontalHeaderLabels(hLabels);
+    QTableWidgetItem* item;
+    for(int i = 0; i < ui->tableWidget->rowCount(); i++)
+    {
+        for(int j = 0; j < ui->tableWidget->columnCount(); j++)
+        {
+            item = new QTableWidgetItem;
+            if(j == 0)
+            {
+                item->setText(quiz.getCollectionOfQuestions()[i].getQuestion());
+            }
+            if(j == 1)
+            {
+                if(m_answer[i] == quiz.getCollectionOfQuestions()[i].getCorrectAnswer())
+                {
+                    item->setText("Tacan");
+                }
+                else
+                {
+                    item->setText("Netacan");
+                }
+            }
+            ui->tableWidget->setItem(i, j, item);
+        }
+    }
+    ui->tableWidget->horizontalHeader()->setSectionResizeMode(0, QHeaderView::Stretch);
+    ui->tableWidget->verticalHeader()->setSectionResizeMode(QHeaderView::Stretch);
+
+    //delete item;    gdje da pozovem ovo?
+
+    QSqlQuery insertIntoHighscoreQuery;
+    insertIntoHighscoreQuery.prepare("INSERT INTO highscore VALUES ((:playerName), (:quizId), (:points))");
+    insertIntoHighscoreQuery.bindValue(":playerName", m_playerName);
+    insertIntoHighscoreQuery.bindValue(":quizId", quiz.getQuizId());
+    insertIntoHighscoreQuery.bindValue(":points", m_correctAnswers);
+
+    insertIntoHighscoreQuery.exec();
+}
+
+void Selection::nextStep(int step)
+{
+    m_answer[indexOfCurrentQuestion] = ui->comboBox_answers->currentIndex() + 1;
+    if(step == -1)
+    {
+        indexOfCurrentQuestion--;
+    }
+    else if(step == 1)
+    {
+        indexOfCurrentQuestion++;
+    }
+    emit Selection::play();
+}
+
+
+void Selection::on_pushButton_finish_clicked()
+{
+    m_answer[indexOfCurrentQuestion] = ui->comboBox_answers->currentIndex() + 1;
+
+    bool check = false;
+    for(int k = 0; k < m_size; k++)
+    {
+        if(m_answer[k] == 0)
+        {
+            check = true;
```

Comment:
Ispod ove linije može break naredba, da ne bi program džabe prolazio kroz ceo niz m_answer (na primer, ispostavi se odmah da na prvi nije odgovoreno, što bismo prolazili za druge?).